pkg/apis/pipeline/v1alpha1/type.go

```diff
@@ -46,10 +46,32 @@ type PipelineSpec struct {
 	// The tasks will be executed in the order they are listed.
 	Tasks []PipelineTask `json:"tasks"`
 
-	// SharedWorkspace is the name of the PVC. If not specified, a PVC with the Pipeline's name as prefix will be created by default.
+	// SharedWorkspace is the config of the PVC where task using
+	// The PersistentVolumeClaim with this config will be created for each pipeline execution
+	// it allows the user to specify e.g. size and StorageClass for the volume.
 	// If not set, Kurator will create a PVC named Pipeline.name using default config
 	// +optional
-	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+	SharedWorkspace *VolumeClaimTemplate `json:"sharedWorkspace,omitempty"`
+}
+
+// VolumeClaimTemplate is the configuration for the volume claim template in pipeline execution.
+// For more details, see https://github.com/kubernetes/api/blob/master/core/v1/types.go
+type VolumeClaimTemplate struct {
+	// AccessMode determines the access modes for the volume, e.g., ReadWriteOnce.
+	// This affects how the volume can be mounted.
+	AccessMode string `json:"accessMode,omitempty"`
+
+	// RequestsStorage defines the storage size required for this PVC, e.g., 1Gi, 100Mi.
+	// It specifies the storage capacity needed as part of ResourceRequirements.
+	RequestsStorage string `json:"requestsStorage,omitempty"`
```

Comment:
s/RequestsStorage/StorageRequest, it is singular

Reply:
ok